test/e2e/specs/editor/blocks/navigation.spec.js

```diff
@@ -315,277 +315,445 @@ test.describe( 'Navigation block', () => {
+
+			/**
+			 * Test: We don't lose focus when closing the submenu appender
+			 */
+
+			// Move focus to the submenu navigation appender
+			await page.keyboard.press( 'End' );
+			await pageUtils.pressKeys( 'ArrowRight', { times: 2 } );
+			await navigation.useBlockInserter();
+			await navigation.addLinkClose();
+			/**
+			 * TODO: This is not desired behavior. Ideally the
+			 * Appender should be focused again since it opened
+			 * the link control.
+			 * IMPORTANT: This check is not to enforce this behavior,
+			 * but to make sure focus is kept nearby until we are able
+			 * to send focus to the appender. It is falling back to the previous sibling.
+			 */
+			await navigation.checkLabelFocus( 'Dog' );
+		} );
 
 		/**
-		 * Test: We can open and close the preview with the keyboard and escape
-		 *       buttons from a top-level nav link with a url-like label using
-		 *       both the shortcut and toolbar
+		 * New test: Use the submenu nav item appender to add a custom link
 		 */
-		await pageUtils.pressKeys( 'ArrowLeft' );
-		await navigation.useLinkShortcut();
-		await navigation.previewIsOpenAndCloses();
-		await navigation.checkLabelFocus( 'example.com' );
-
-		await navigation.canUseToolbarLink();
+		// eslint-disable-next-line playwright/expect-expect
+		test( 'Can add submenu item(custom-link) using the keyboard', async ( {
+			page,
+			pageUtils,
+			navigation,
+			editor,
+		} ) => {
+			await pageUtils.pressKeys( 'ArrowDown' );
+			await navigation.useBlockInserter();
+			await navigation.addPage( 'Cat' );
 
-		/**
-		 * Test: Can add submenu item using the keyboard
-		 */
-		navigation.useToolbarButton( 'Add submenu' );
+			/**
+			 * Test: Can add submenu item using the keyboard
+			 */
+			navigation.useToolbarButton( 'Add submenu' );
 
-		// Expect the submenu Add link to be present
-		await expect(
-			editor.canvas.locator( 'a' ).filter( { hasText: 'Add link' } )
-		).toBeVisible();
+			// Expect the submenu Add link to be present
+			await expect(
+				editor.canvas.locator( 'a' ).filter( { hasText: 'Add link' } )
+			).toBeVisible();
 
-		await pageUtils.pressKeys( 'ArrowDown' );
-		// There is a bug that won't allow us to press Enter to add the link: https://github.com/WordPress/gutenberg/issues/60051
-		// TODO: Use Enter after that bug is resolved
-		await navigation.useLinkShortcut();
+			await pageUtils.pressKeys( 'ArrowDown' );
+			// There is a bug that won't allow us to press Enter to add the link: https://github.com/WordPress/gutenberg/issues/60051
+			// TODO: Use Enter after that bug is resolved
+			await navigation.useLinkShortcut();
+
+			await navigation.addPage( 'Dog' );
+
+			/**
+			 * Test: We can open and close the preview with the keyboard and escape
+			 *       buttons from a submenu nav item using both the shortcut and toolbar
+			 */
+			await navigation.useLinkShortcut();
+			await navigation.previewIsOpenAndCloses();
+			await navigation.checkLabelFocus( 'Dog' );
```

Comment:
This seems redundant since it was already tested in the previous test.

Perhaps doing just `addPage( 'Cat' )`, `addPage( 'Dog' )` would be enough?